photon-api/src/main/scala/com/linkedin/photon/ml/util/MathUtils.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.util
+
+import com.linkedin.photon.ml.constants.MathConst
+
+/**
+ *
+ */
+object MathUtils {
+
+  /**
+   * This function is copied from MLlib's MLUtils.log1pExp.
```

Comment:
Link, for the skeptical: [Line 527](https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/util/MLUtils.scala)